plugins/lime-plugin-fbw/src/containers/NetworkForm.js

```diff
@@ -12,48 +12,72 @@ import I18n from 'i18n-js';
+	const [state, setState] = useState({
 		communityName: '',
 		hostName: '',
-		password: ''
+		password: '',
+		passwordConfirmation: '',
 	});
 
-	function _changeName (e) {
+	function _changeName(e) {
 		const end = e.type === 'change';
 		e.target.value = slugify(e.target.value.toLocaleLowerCase(), end, true);
 		setState({ ...state, communityName: e.target.value || '' });
 		return e;
 	}
 
-	function _changeHostName (e) {
+	function _changeHostName(e) {
 		const end = e.type === 'change';
 		e.target.value = slugify(e.target.value, end);
 		setState({ ...state, hostName: e.target.value || '' });
 		return e;
 	}
 
-	/* function _changePassword (e){
-		setState({ ...stat, password: e.target.value || '' });
-	} */
+	function _changePassword(e) {
+		setState({ ...state, password: e.target.value || '' });
+		return e;
+	}
+
+	function _changePasswordConfirmation(e) {
+		setState({ ...state, passwordConfirmation: e.target.value || '' });
+		return e;
+	}
 
-	function _createNetwork(){
-		createNetwork({ network: state.communityName, hostname: state.hostName });
+	function _createNetwork() {
+		createNetwork({
+			network: state.communityName,
+			hostname: state.hostName,
+			password: state.password,
+		});
 		toggleForm('setting')();
 	}
 
+	function _isValidPassword() {
+		return state.password.length > 0;
+	}
+
+	function _passwordMissMatch() {
+		return state.password && state.passwordConfirmation && state.password !== state.passwordConfirmation;
+	}
+
 	return (<div class="container" style={{ paddingTop: '100px' }}>
 		<h4><span>{I18n.t('Configure your new community network')}</span></h4>
 		<label>{I18n.t('Choose a name for your network')}</label>
-		<input type="text" placeholder={I18n.t('Community name')} class="u-full-width" onChange={_changeName} onInput={_changeName} />
+		<input type="text" placeholder={I18n.t('Community name')} class="u-full-width" onInput={_changeName} />
+		<label>{I18n.t('Choose a shared password for network adminitration')}</label>
+		<input type="password" placeholder={I18n.t('Password')} class="u-full-width" value={state.password} onInput={_changePassword} />
+		<label>{I18n.t('Re-enter the shared password')}</label>
+		<input type="password" placeholder={I18n.t('Re-enter Password')} class="u-full-width" value={state.passwordConfirmation} onInput={_changePasswordConfirmation} />
+		{_passwordMissMatch() &&
+			<label>{I18n.t('The passwords do not match!')}</label>
+		}
 		<label>{I18n.t('Choose a name for this node')}</label>
-		<input type="text" placeholder={I18n.t('Host name')} class="u-full-width" value={state.hostName} onChange={_changeHostName} onInput={_changeHostName} />
-		{/* <label>{I18n.t('Choose a password for this node')}</label>
-		<input type="text" placeholder={I18n.t('Password')} class="u-full-width" value={state.password} onChange={_changePassword} /> */}
+		<input type="text" placeholder={I18n.t('Host name')} class="u-full-width" value={state.hostName} onInput={_changeHostName} />
```

Comment:
I am not sure from the code, network, and hostname and password all required form values?

Reply:
it wasn't neither clear nor correct, thank you! Check it now :)